cmd/fabric8-jenkins-proxy/main.go

```diff
@@ -227,3 +214,25 @@ func setupSignalChannel(cancel context.CancelFunc) {
 		cancel()
 	}()
 }
+
+func getAPIServer(api api.ProxyAPI) *http.Server {
+	c := cors.New(cors.Options{
+		AllowCredentials: true,
+	})
+	srv := &http.Server{
+		Addr:    apiRouterPort,
+		Handler: c.Handler(router.CreateAPIRouter(api)),
+	}
+	return srv
+}
+
+func getProxyServer(p *proxy.Proxy) *http.Server {
+	c := cors.New(cors.Options{
```

Comment:
I am not sure if we need cors for the reverse proxy. Could you please explain why this is needed?

Reply:
Just kept what was there in the previous commit I guess? :)

Reply:
@sbose78 aha, that is what I suspected. So in general, we don't need to CORS for reverse proxy side and only need it for the api server which is the part responsible for providing the `state` and the `unidle` api. We would also need to expose this through a route different route e.g. api.jenkins.openshift.io or jenkins.openshift.io/proxy/api/ since `/api/` IIUC is already an endpoint exposed by the actual jenkins.

Reply:
> We would also need to expose this through a route different route e.g. api.jenkins.openshift.io or jenkins.openshift.io/proxy/api/ since /api/ IIUC is already an endpoint exposed by the actual jenkins.

Absolutely agreed @sthaha . 
but please do check if it's OK to expose the API route, *in case* there was a good reason for not exposing it in the first place.